### PR TITLE
CVS-68969 Disable caching by default for models with custom loaders and shape/batch auto

### PR DESCRIPTION
**Pull request by dkalinowski**

### CODE REVIEW

File: src/test/modelconfig_test.cpp [hunk 1]
```diff
@@ -656,6 +656,198 @@ TEST(ModelConfig, ConfigParseNodeWithValidShapeFormatArray) {
...
+
+    const auto modelConfigList = configJson.FindMember("model_config_list");
+    ASSERT_NE(modelConfigList, configJson.MemberEnd());
+    const auto& configs = modelConfigList->value.GetArray();
+    ASSERT_EQ(configs.Size(), 1);
+    ovms::ModelConfig modelConfig;
+    auto status = modelConfig.parseNode(configs[0]["config"]);
+
+    ASSERT_EQ(status, ovms::StatusCode::OK);
+    EXPECT_TRUE(modelConfig.isCachingDisabled());
```
Comment (atobiszei) [inline src/test/modelconfig_test.cpp:794]: I think that we shouldn't at all load the model when the configuration is wrong and at least log the error.

Comment (dkalinowski) [reply src/test/modelconfig_test.cpp:794]: We log model cache setting in debug mode:
https://github.com/openvinotoolkit/model_server/blob/develop/src/modelinstance.cpp#L508
@atobiszei is it enough?

File: docs/model_cache.md [hunk 2]
```diff
@@ -26,18 +26,19 @@ When using Model Server with configuration file, it is possible to serve more th
...
         {"config": {
             "name": "face_detection",
             "base_path": "/models/face-detection-retail-0004/",
             "shape": "auto",
-            "force_caching": true}},
+            "allow_cache": true}},
     ],
 }
 ```
+> IMPORTANT: It is not possible to force enablement of cache for models with custom loaders.
```
Comment (dtrawins) [inline docs/model_cache.md:41]:
```suggestion
> IMPORTANT: Models imported via the custom loaders never create or use any cache.
```

File: docs/model_cache.md [hunk 3]
```diff
@@ -26,15 +26,15 @@ When using Model Server with configuration file, it is possible to serve more th
 - Models with custom loader (for security reasons explained earlier)
 - Models configured to shape `auto` or batch_size `auto`
 
-In case there are valid reasons to enable the model cache also in those two conditions, it is possible to force enablement with `"force_caching": true` parameter:
+In case there are valid reasons to enable the model cache also in those two conditions, it is possible to force enablement with `"allow_cache": true` parameter:
```
[the remark this answers is not in the document]
Comment (dkalinowski) [reply docs/model_cache.md]: Done

**Review by ravikumarbhattiprolu — APPROVED** [review]

File: src/test/modelconfig_test.cpp (hunk 1 above)
Comment (dkalinowski) [reply src/test/modelconfig_test.cpp:794]: After sync we agreed this is not enough. Agreed also to merge this and treat this as bug, ticket: https://jira.devtools.intel.com/browse/CVS-71598

File: src/modelconfig.cpp [hunk 4]
```diff
@@ -611,11 +609,23 @@ Status ModelConfig::parseNode(const rapidjson::Value& v) {
...
+        this->setDisableCaching(!v["allow_cache"].GetBool());
+        SPDLOG_DEBUG("allow_cache: {}", !isCachingDisabled());
+    }
+
     // if the config has models which require custom loader to be used, then load the same here
     if (v.HasMember("custom_loader_options")) {
         if (!parseCustomLoaderOptionsConfig(v["custom_loader_options"]).ok()) {
             SPDLOG_ERROR("Couldn't parse custom loader options config");
         }
+        // force disable caching
```
[the remark this answers is not in the document]
Comment (dkalinowski) [reply src/modelconfig.cpp:627]: If allow caching is specified in config file along with customloader, the only sign of warning is log in log_level=DEBUG mode saying that "Model was not loaded from cache". There is separate bug tracked to make the model not load at all and set LOADING, ERROR model status.
https://jira.devtools.intel.com/browse/CVS-71598

@ravikumarbhattiprolu what is your suggestion?

[the remark this answers is not in the document]
Comment (ravikumarbhattiprolu) [reply src/modelconfig.cpp:627]: Better show it as error and avoid loading the model. This can be treated as config error

**Review by atobiszei — APPROVED** [review]

**Pull request merged**